File: kotlin/4.kt
fun String.titleCase(delimiter: String = " ") =
    split(delimiter).joinToString(delimiter) { word ->

        val smallCaseWord = word.lowercase()
        smallCaseWord.replaceFirstChar(Char::titlecaseChar)

    }

    fun String.swapcase(): String {
        var resultTemp = ""
        for (char in this) {
            resultTemp += if (char.isLowerCase()) char.uppercase() else char.lowercase()
        }
        return resultTemp
    }
fun main(){
    val example = "this IS a stRing";
    println(example.uppercase());
    println(example.lowercase());
    println(example.titleCase()); // this is using an extension function 
    println(example.swapcase());
    println(example.replaceFirstChar { transform -> transform.uppercase() }); // this is using a lambda function
}

File: js/4.js
var example;
example = "this IS a stRing";
console.log(example.toLowerCase());
console.log(example.toUpperCase());
console.log(titleCase(example));
console.log(swapCase(example));
console.log(capitalize(example));

function titleCase(str) {
    str = str.toLowerCase();
    str = str.split(' ');
    for (var i = 0; i < str.length; i++) {
      str[i] = str[i].charAt(0).toUpperCase() + str[i].slice(1); 
    }
    return str.join(' '); 
  }
function swapCase (str){
    var newstr = "";
    for(var i = 0; i<str.length; i++){
        if(str[i] === str[i].toLowerCase()){
            newstr += str[i].toUpperCase();
        }else {
            newstr += str[i].toLowerCase();
        }
    }
    console.log(newstr);
    return newstr;
}
function capitalize(str) {
    return str.charAt(0).toUpperCase() + str.slice(1);
}